### PR TITLE
Student][Teacher][MBL-12116] Add course color overlay toggle

### PR DESCRIPTION
**Pull request by JordanMarshall**

### CODE REVIEW

File: apps/student/src/main/res/layout/navigation_drawer.xml [hunk 1]
```diff
@@ -217,6 +217,34 @@
...
+                    android:padding="1dp"
+                    android:tint="@color/defaultTextDark"
+                    app:srcCompat="@drawable/vd_courses"/>
+
+                <TextView
+                    style="@style/TextFont.Medium"
+                    android:layout_width="0dp"
+                    android:layout_height="wrap_content"
+                    android:layout_weight="1"
+                    android:labelFor="@+id/navigationDrawerShowGradesSwitch"
```
Comment (CalvinKern) [inline apps/student/src/main/res/layout/navigation_drawer.xml]: This is actually a label for `navigationDrawerColorOverlaySwitch`

File: apps/teacher/src/main/res/layout/navigation_drawer.xml [hunk 2]
```diff
@@ -172,6 +172,31 @@
...
+                    style="@style/NavigationDrawerIcon"
+                    app:srcCompat="@drawable/vd_courses"/>
+
+                <TextView
+                    style="@style/TextFont.Medium"
+                    android:layout_width="0dp"
+                    android:layout_height="wrap_content"
+                    android:layout_weight="1"
+                    android:text="@string/colorOverlay"
+                    android:textSize="16sp"/>
```
Comment (CalvinKern) [inline apps/teacher/src/main/res/layout/navigation_drawer.xml]: Could add a `labelFor` navigationDrawerColorOverlaySwitch to make accessibility a little nicer.

File: libs/canvas-api-2/src/main/java/com/instructure/canvasapi2/CanvasRestAdapter.kt [hunk 3]
```diff
@@ -326,7 +326,11 @@ protected constructor(var statusCallback: StatusCallback<*>?, private val authUs
                 val regex = Regex(pattern)
                 val urls = okHttpClient.cache()?.urls() ?: return
                 while (urls.hasNext()) {
-                    if (regex.containsMatchIn(urls.next())) urls.remove()
+                    val next = urls.next()
+                    if (regex.containsMatchIn(next)) {
+                        urls.remove()
+                        Logger.d("Clearing cached url: $next")
```
Comment (CalvinKern) [inline libs/canvas-api-2/src/main/java/com/instructure/canvasapi2/CanvasRestAdapter.kt]: Did you want to leave this logging in?

Comment (JordanMarshall) [reply libs/canvas-api-2/src/main/java/com/instructure/canvasapi2/CanvasRestAdapter.kt]: Yeah, I think it will be useful during development.

**Pull request merged**